### PR TITLE
py-rdflib: add 6.3.2

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/RDFLib/rdflib/tree/6.3.2

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the version sha256.

File: var/spack/repos/builtin/packages/py-rdflib/package.py [hunk 1]
```diff
@@ -21,15 +21,23 @@ class PyRdflib(PythonPackage):
...
     version("6.2.0", sha256="62dc3c86d1712db0f55785baf8047f63731fa59b2682be03219cb89262065942")
     version("6.0.2", sha256="6136ae056001474ee2aff5fc5b956e62a11c3a9c66bb0f3d9c0aaa5fbb56854e")
     version("5.0.0", sha256="78149dd49d385efec3b3adfbd61c87afaf1281c30d3fcaf1b323b34f603fb155")
 
-    depends_on("python@3.7:", when="@6:", type="build")
-    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:3", when="@6.3:", type=("build", "run"))
+    depends_on("py-poetry-core@1.4:", when="@6.3:", type="build")
 
+    depends_on("py-isodate@0.6:0", when="@6.3:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rdflib/package.py]:
```suggestion
    depends_on("py-isodate@0.6", when="@6.3:", type=("build", "run"))
```
See https://python-poetry.org/docs/dependency-specification/#caret-requirements

File: var/spack/repos/builtin/packages/py-rdflib/package.py [hunk 2]
```diff
@@ -21,15 +21,23 @@ class PyRdflib(PythonPackage):
...
     depends_on("py-isodate", type=("build", "run"))
+    depends_on("py-pyparsing@2.1:3", when="@6.3:", type=("build", "run"))
     depends_on("py-pyparsing", type=("build", "run"))
-    depends_on("py-setuptools", when="@6:", type=("build", "run"))
+    depends_on("py-importlib-metadata@4", when="@6.3: ^python@:3.7", type=("build", "run"))
     depends_on("py-importlib-metadata", when="@6.1: ^python@:3.7", type=("build", "run"))
+
+    # Historical dependencies
+    depends_on("python@3.7:", when="@6:6.2", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rdflib/package.py]: Could remove this line

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-rdflib/package.py (hunk 1 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-rdflib/package.py]: Thanks

File: var/spack/repos/builtin/packages/py-rdflib/package.py (hunk 2 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-rdflib/package.py]: Done

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**